### ISSUE TITLE
Binary FMA test fails with C/R

### ISSUE THREAD

**Issue by vkbo** [SixTrack/SixTrack#626]
When run with checkpoint restarting, the singletrackfile is truncated.

**Comment by kyrsjo** [comment]
It seems to affect most of the big FMA tests -- fma_binary, dump7_fma, ... it seems that all the dumps and the singletrackfile are truncated. fort.10 is also not matching.

It seems to run for a few restarts, but then crashing with `errout_status=-1`, the first error messages are:
```
******* Last     6 lines of file 'fort.93': *******
 SIXTRACR STOP/ABEND checking fort.10
 SIXTRACR STOP/ABEND writing a fort.10
 SIXTRACR STOP/ABEND writing fort.10, lines           0 /          30
 SIXTRACR STOP/ABEND copying fort.92 to fort.6
 SIXTRACR stop

******* Done writing tail of file 'fort.93' to stderr *******
******* Last     2 lines of file 'fort.6': *******
ABEND> ERROR Problems writing to fort.10. ierro = 5001
 SIXTRACR stop
******* Done writing tail of file 'fort.6' to stderr *******
Stopping, errout_status=   -1
STOP 1
SixTrack CR was killed: 256
```

Later it crashes like:
```
******* Last     4 lines of file 'fort.93': *******
 SIXTRACR STOP/ABEND checking fort.10
 SIXTRACR STOP/ABEND copying fort.92 to fort.6
 SIXTRACR stop

******* Done writing tail of file 'fort.93' to stderr *******
******* Last    40 lines of file 'fort.6': *******
  Number of reverse-revolutions:                            0
  Turns per coor.-printout:                             50000
  Flat bottom up to turn:                                   0
  Turns per print on dataset:                               1
  Acceleration up to turn:                                  0
  Turns per print on dataset:                               1
  Flat top number of turns:                              5000

  Turns per print on dataset:                               1
  Tracking start at element no.:                            1
  Initial amplitude-h in (mm):                              0.489704000
  Coupling  eps-y/eps-x:                                    0.071797000
  Number of C.O. iterations:                               50
  Precision of C.O. deviation:                              0.100000000E-11
  Precision of C.O. slope:                                  0.100000000E-14
  Number of q-adj. iterations:                             10
  Change in k-strength by:                                  0.100000000E-09
  Precision of q-adjustement:                               0.100000000E-09
  Number of chromat.-adj. iter.:                           10
  Change in sex.-strength by:                               0.100000000E-05
  Precision of chromat.-adj.:                               0.100000000E-05
  DP-interval f. cromat.-adj.:                              0.100000000E-05
  DP-interval for dispersion:                               0.100000000E-08
  Precision for C.O. RMS:                                   0.100000000E-08

------------------------------------------------------------------------------------------------------------------------------------

    OOOOOOOOOOOOOOOOOOOOO
    OO                 OO
    OO  PREPROCESSING  OO
    OO                 OO
    OOOOOOOOOOOOOOOOOOOOO

------------------------------------------------------------------------------------------------------------------------------------
SIXTRACR> ERROR DUMP_CRCHECK_POSITIONFILES failure positioning DUMP file
    +++++++++++++++++++++++++++++
    +      ERROR DETECTED       +
    + RUN TERMINATED ABNORMALLY +
    +++++++++++++++++++++++++++++
 SIXTRACR stop
******* Done writing tail of file 'fort.6' to stderr *******
Stopping, errout_status=   -1
STOP 1
SixTrack CR was killed: 256
```

Does the line `ABEND> ERROR Problems writing to fort.10. ierro = 5001` tell you something, @vkbo ?

Furthermore, a separate issue is that the message `SIXTRACR> ERROR DUMP_CRCHECK_POSITIONFILES failure positioning DUMP file` does not give sufficient information to debug, and while the information (sans the actual filename) should be in fort.93, it isn't.

Could this be related to the new fileunits functionality?

**Comment by vkbo** [comment]
I only saw it for fma_binary, not the others. I think that confirms our suspicion that it is killed in the middle of the FMA postprocessing routine.

**Comment by vkbo** [comment]
Re: the fileunits thing. No. This is unrelated. The fileunits module only handles units for non-fort.x files.
The mod_units module is a wrapper for open() that will account for BOINC, etc, but that's all it does.

**Comment by vkbo** [comment]
Fixed in PR #629
